packages/icon/test/icon.test.js

```diff
@@ -98,6 +98,38 @@ describe('vaadin-icon', () => {
     });
   });
 
+  describe('src property', () => {
+    beforeEach(() => {
+      icon = fixtureSync('<vaadin-icon></vaadin-icon>');
+      svgElement = icon.shadowRoot.querySelector('svg');
+    });
+
+    it('should render SVG when path is provided', async () => {
+      const svgSrc = `<svg>${ANGLE_DOWN}</svg>`;
+      const fakeFetch = () => Promise.resolve(svgSrc);
+      icon.__fetchSVG = fakeFetch;
+      icon.src = `data:image/svg+xml,${encodeURIComponent(svgSrc)}`;
+      await fakeFetch();
+      expectIcon(ANGLE_DOWN);
+    });
+
+    it('should fail if SVG is not found', async () => {
```

Comment:
This test seems to be flaky. It fails randomly when I re-run the tests

<img width="573" alt="Screenshot 2023-08-22 at 9 12 41" src="https://github.com/vaadin/web-components/assets/1222264/489c67b1-f6a3-4a08-9e18-0a07b7279c43">